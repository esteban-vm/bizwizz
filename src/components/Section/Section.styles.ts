import clsx from 'clsx'

const styles = {
  wrapper(bg: 'white' | 'black') {
    return clsx('pb-32 text-center', bg === 'white' ? 'bg-white' : 'bg-black')
  },

  header: clsx('font-bold tracking-widest'),
  heading: clsx('mb-5 text-[2.4rem] uppercase text-primary'),
  subheading: clsx('mb-10 text-[4rem] capitalize text-darkgrey'),
}

export default styles
