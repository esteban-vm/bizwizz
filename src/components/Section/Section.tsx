import type { FC, PropsWithChildren } from '@/types'
import { Wave } from '../Wave'
import styles from './Section.styles'

type SectionProps = PropsWithChildren<{
  id: Lowercase<string>
  title: Capitalize<string>
  subtitle?: Capitalize<string>
  bg: 'white' | 'black'
}>

const Section: FC<SectionProps> = ({ id, title, subtitle, bg, children }) => {
  const titleID = `${id}_title`

  return (
    <section aria-labelledby={titleID} className={styles.wrapper(bg)} id={id}>
      <Wave is={bg === 'black' ? 'white' : 'black'} />
      <header className={styles.header}>
        <h2 className={styles.heading} id={titleID}>
          {title}
        </h2>
        {subtitle && <h3 className={styles.subheading}>{subtitle}</h3>}
      </header>
      {children}
    </section>
  )
}

export default Section
