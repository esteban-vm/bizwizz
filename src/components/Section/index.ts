export { default as Section, type SectionID } from './Section'
