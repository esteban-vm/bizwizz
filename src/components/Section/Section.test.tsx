import { render, cleanup, within, screen } from '@/tests'
import Section from './Section'

describe('Section:', () => {
  let asFragment: () => DocumentFragment

  beforeEach(() => {
    void ({ asFragment } = render(
      <Section bg='black' id='test' subtitle='Lorem ipsum' title='Test'>
        <article>Child element</article>
      </Section>
    ))
  })

  afterEach(cleanup)

  it('should render correctly', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it('should have correct structure', () => {
    const title = /Test/
    const subtitle = /Lorem ipsum/
    const sectionElem = screen.getByRole('region')
    expect(sectionElem).toBeInTheDocument()
    expect(sectionElem).toHaveAccessibleName(title)
    expect(sectionElem).toHaveAttribute('id', 'test')
    expect(within(sectionElem).getByRole('presentation', { hidden: true })).toBeInTheDocument()
    expect(within(sectionElem).getByRole('heading', { name: title, level: 2 })).toBeInTheDocument()
    expect(within(sectionElem).getByRole('heading', { name: subtitle, level: 3 })).toBeInTheDocument()
    expect(within(sectionElem).getByRole('article')).toBeInTheDocument()
    expect(within(sectionElem).getByRole('article')).toHaveTextContent(/child element/i)
  })
})
