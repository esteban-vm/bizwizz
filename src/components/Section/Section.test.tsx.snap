// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Section: > should render correctly 1`] = `
<DocumentFragment>
  <section
    aria-labelledby="test_title"
    class="pb-32 text-center bg-black"
    id="test"
  >
    <svg
      aria-hidden="true"
      class="h-40 w-full text-white"
      preserveAspectRatio="none"
      role="presentation"
      viewBox="0 0 500 150"
    >
      <path
        d="M0.00,49.98 C161.96,151.48 356.65,-54.75 500.00,49.98 L504.51,-36.00 L-13.54,-3.44 Z"
        fill="currentColor"
        stroke="none"
      />
    </svg>
    <header
      class="font-bold tracking-widest"
    >
      <h2
        class="mb-5 text-[2.4rem] uppercase text-primary"
        id="test_title"
      >
        Test
      </h2>
      <h3
        class="mb-10 text-[4rem] capitalize text-darkgrey"
      >
        Lorem ipsum
      </h3>
    </header>
    <article>
      Child element
    </article>
  </section>
</DocumentFragment>
`;
